src/main/java/io/netty/incubator/codec/http3/Http3PushStreamInitializer.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.incubator.codec.http3;
+
+import io.netty.channel.ChannelInitializer;
+import io.netty.incubator.codec.quic.QuicStreamChannel;
+
+import static io.netty.incubator.codec.quic.QuicStreamType.BIDIRECTIONAL;
+
+abstract class Http3PushStreamInitializer extends ChannelInitializer<QuicStreamChannel> {
+
+    static void verifyIsUnidirectional(QuicStreamChannel ch) {
+        if (ch.type() == BIDIRECTIONAL) {
```

Comment:
I think I would use `ch.type() != UNIDIRECTIONAL` ... This ensures we will not get into trouble if the enum would receive another type at some point.. That said I cant think of why it would but better safe then sorry :)